src/ACEtk/block/SecondaryParticleLocatorBlock/src/ctor.hpp

```diff
@@ -0,0 +1,42 @@
+private:
+
+/**
+ *  @brief Private constructor
+ */
+SecondaryParticleLocatorBlock(
+    std::vector< std::array< unsigned int, 10 > >&& locators,
+    std::size_t ntype ) :
+  Base( "IXS", generateXSS( std::move( locators ) ) ),
+  ntype_( ntype ) {}
+
+public:
+
+SecondaryParticleLocatorBlock() = default;
+
+SecondaryParticleLocatorBlock( const SecondaryParticleLocatorBlock& ) = default;
+SecondaryParticleLocatorBlock( SecondaryParticleLocatorBlock&& ) = default;
+
+/**
+ *  @brief Constructor
+ *
+ *  @param[in] locators    the locators for each secondary particle type
+ */
+SecondaryParticleLocatorBlock(
+    std::vector< std::array< unsigned int, 10 > > locators ) :
+  SecondaryParticleLocatorBlock( std::move( locators ), locators.size() ) {}
```

Comment:
Use-after-move (I think)
I'm going to pause checking these until a way forward is decided, because there's quite a few of them.